[nativert] Improve MPMCQueue tests.

zhxchen17 opened the pull request:
Summary:
- Use std::this_thread::yield and stop busy wating.
- Sort test file orders.

Following up @swolchok's comment from https://github.com/pytorch/pytorch/pull/152837
Test Plan: CI

Differential Revision: D74402536

cyyever commented on test/cpp/nativert/CMakeLists.txt:
```diff
@@ -2,8 +2,8 @@ set(NATIVERT_TEST_ROOT ${TORCH_ROOT}/test/cpp/nativert)
 set(NATIVERT_TEST_SRCS
-  ${NATIVERT_TEST_ROOT}/test_tensor_meta.cpp
-  ${NATIVERT_TEST_ROOT}/test_mpmc_queue.cpp
+${NATIVERT_TEST_ROOT}/test_mpmc_queue.cpp
```
Should indent.

cyyever commented on test/cpp/nativert/test_mpmc_queue.cpp:
```diff
@@ -53,6 +53,7 @@ TEST(MPMCQueueTest, ConcurrentAccess) {
       int out = 0;
       while (!queue.readIfNotEmpty(out)) {
         // Wait until an element is available
+        std::this_thread::yield();
```
Does the queue have a method to wait until read available? Use spinning wait is quite strange.

zhxchen17 replied:
Yeah but we don't provide a blocking wait for now. Internally we use a semaphore paired with this (not upstreamed yet)

cyyever commented on test/cpp/nativert/test_mpmc_queue.cpp:
```diff
@@ -76,6 +77,7 @@ TEST(MPMCQueueTest, MPMCConcurrentAccess) {
         size_t value = i * numElementsPerWriter + j;
         while (!queue.writeIfNotFull(static_cast<int>(value))) {
           // Retry until the queue has space
+          std::this_thread::yield();
```
Same as above, it should provide wait until having data.

cyyever commented on test/cpp/nativert/test_mpmc_queue.cpp:
```diff
@@ -90,6 +92,8 @@ TEST(MPMCQueueTest, MPMCConcurrentAccess) {
         if (queue.readIfNotEmpty(value)) {
           ++totalReadCount;
+        } else {
+          std::this_thread::yield();
```
Same

cyyever reviewed (approved):
LGTM, but consider the comments.

zhxchen17 committed:
[nativert] Improve MPMCQueue tests. (#153154)

Summary:

- Use std::this_thread::yield and stop busy wating.
- Sort test file orders.

Test Plan: CI

Reviewed By: dolpm

Differential Revision: D74402536
